libpoolprotocols/PoolManager.h

```diff
@@ -16,35 +17,38 @@ namespace dev
 {
 	namespace eth
 	{
-		class PoolManager : public Worker
+		class PoolManager
 		{
 		public:
-			PoolManager(PoolClient * client, Farm &farm, MinerType const & minerType);
+			PoolManager(PoolClient * client, Farm &farm, MinerType const & minerType, unsigned maxTries);
 			void addConnection(URI &conn);
 			void clearConnections();
 			void start();
 			void stop();
-			void setReconnectTries(unsigned const & reconnectTries) { m_reconnectTries = reconnectTries; };
 			bool isConnected() { return p_client->isConnected(); };
 			bool isRunning() { return m_running; };
 
 		private:
 			unsigned m_hashrateReportingTime = 60;
 			unsigned m_hashrateReportingTimePassed = 0;
 
-			bool m_running = false;
-			void workLoop() override;
-			unsigned m_reconnectTries = 3;
-			unsigned m_reconnectTry = 0;
-			std::vector <URI> m_connections;
+			std::atomic<bool> m_running = { false };
```

Comment:
There is also `g_running` flag. Can this be combined?

Reply:
Combining global running semaphore (which is only affected by signal capturing) with m_running on PoolManager requires a sensible redesign of running pattern and, at least, remove (on PoolManager) the start and stop methods as they would make no sense.
Nevertheless actual separation of running ethminer process from PoolManager service may (or may not) be helpful to implement further features like, say for example, bind to recent added feature --tstart/--tstop where poolconnection might be suspended in case **all** gpu exceed temperature threshold.

Anyway I would leave this task outside the scope of **this** specific PR.